### DISCUSSION TITLE
Enhancement Request - Connect to Remote Machines

### DISCUSSION THREAD

**Discussion by spanningtrees42** [Timthreetwelve/WUView#116] [Ideas]
### Brief summary of your request

Add Load From option?

### Request details

Are there plans to add a Load From option?  This would be very useful to run on remote machines.

**Comment by Timthreetwelve** [comment]
@spanningtrees42, I would love it If I could make this happen, I can see how it would be extremely useful. I've investigated this previously, and I don't see a straightforward way to fulfill your request.

WUView uses the **Windows Update Agent API** (WUApiLib.dll) to get the list of updates. From what I can tell, there isn't a way to point it to a specific file, drive, or machine. It just gets the updates from the machine that it's on.

If a way to connect to another machine presents itself, I will definitely see if I can incorporate it into WUView.

Thanks,
Tim

**Comment by spanningtrees42** [comment]
> @spanningtrees42, I would love it If I could make this happen, I can see how it would be extremely useful. I've investigated this previously, and I don't see a straightforward way to fulfill your request.
> 
> WUView uses the **Windows Update Agent API** (WUApiLib.dll) to get the list of updates. From what I can tell, there isn't a way to point it to a specific file, drive, or machine. It just gets the updates from the machine that it's on.
> 
> If a way to connect to another machine presents itself, I will definitely see if I can incorporate it into WUView.
> 
> Thanks, Tim

There is similar product from Nirsoft, WinUpdatesView, I don't know how different the app arch is from yours but you may can see how it functions.  It has the following options to retrieve updates:

Specifies the type of data source to load:
1 - Local computer, with API
2 - Local computer, Read database file directly.
3 - Load from external database file.
4 - Load from remote computer.


I really like your app, looks great and runs nicely!

**Comment by Timthreetwelve** [comment]
@spanningtrees42, thanks for pointing out the Nirsoft app. It's a great app, I've used it to check the accuracy of WUView. It appears to read the Windows Update database for options 2-4. I'll do more research to see if there is another API that will facilitate reading the database. In the meantime, I'm concentrating my efforts on making WUView multilingual.

Thanks,
Tim

**Comment by JPOMunster** [comment]
Hi Tim.  I came across your app which i like very much. Just to support @spanningtrees42, having  the ability to query multiple remote Virtual or Physical machines, possibly using AD? and report to a useable format would be very helpful.
Kind Regards
